packages/ubuntu_wizard/lib/src/wizard_button.dart

```diff
@@ -16,131 +16,49 @@ final _noAnimation = Listenable.merge([]);
-          label: UbuntuLocalizations.of(context).previousLabel,
-          visible: visible,
-          flat: true,
-          enabled: !isLoading && (enabled ?? hasPrevious),
-          onActivated: onBack,
-          execute: () {
-            // navigate the root wizard at the end of a nested wizard
-            final effectiveWizard =
-                (wizard?.hasPrevious ?? false) ? wizard : rootWizard;
-            return effectiveWizard?.back();
-          }),
-    );
-  }
-
-  /// A _Next_ button that advances to the next page.
-  static Widget next(
-    BuildContext context, {
-    String? label,
-    bool? visible,
-    bool? enabled,
-    bool? flat,
-    bool? highlighted,
-    Object? arguments,
-    WizardCallback? onNext,
-    WizardCallback? onBack,
-  }) {
-    final wizard = Wizard.maybeOf(context);
-    final rootWizard = Wizard.maybeOf(context, root: true);
-    final routeData =
-        (wizard?.routeData ?? rootWizard?.routeData) as WizardRouteData?;
-    final hasNext = routeData?.hasNext ??
-        ((wizard?.hasNext ?? false) || (rootWizard?.hasNext ?? false));
-    final isLoading =
-        (wizard?.isLoading ?? false) || (rootWizard?.isLoading ?? false);
-    return AnimatedBuilder(
-      animation: wizard?.controller ?? _noAnimation,
-      builder: (context, child) => WizardButton(
-        label: label ??
-            (hasNext
-                ? UbuntuLocalizations.of(context).nextLabel
-                : UbuntuLocalizations.of(context).doneLabel),
-        visible: visible,
-        enabled: !isLoading && (enabled ?? true),
-        loading: isLoading,
-        flat: flat,
-        highlighted: highlighted,
-        onActivated: onNext,
-        execute: () async {
-          // navigate the root wizard at the end of a nested wizard
-          final effectiveWizard =
-              (wizard?.hasNext ?? false) ? wizard : rootWizard;
-          try {
-            await effectiveWizard?.next(arguments: arguments);
-          } on WizardException catch (_) {
-            if (effectiveWizard != rootWizard) {
-              await rootWizard?.next(arguments: arguments);
-            }
-          }
-          onBack?.call();
-        },
-      ),
-    );
-  }
-
   /// Text label of the button.
   final String? label;
 
   /// Determines whether the button is visible.
   ///
   /// The default value is `true`
-  final bool? visible;
+  final bool visible;
 
   /// Determines whether the button is flat (no background).
   ///
   /// The default value is `false`.
-  final bool? flat;
+  final bool flat;
 
   /// Determines whether the button is highlighted (green);
   ///
   /// The default value is `false`.
-  final bool? highlighted;
+  final bool highlighted;
 
   /// Determines whether the button is enabled.
   ///
-  /// The default value is `false`. If not specified, the button is implicitly
+  /// The default value is `true`. If not specified, the button is implicitly
```

Comment:
Just double-checking that this is intended. Has the logic or the semantics changed here?
If the default value for `enabled` is `true` and `onActivated` is `null`, will the button be disabled or enabled?

Reply:
It will be disabled, but the field `enabled` defaults to true. The semantics have changed.
You can see here how it defaulted to true before too:
https://github.com/canonical/ubuntu-desktop-provision/pull/341/files#diff-8ea315d2174db0c87c77ccc3b4f004d1b41357277a429bff6eba7a7b636028a8L95

Reply:
ok, thanks that makes sense. Just a bit slow on evaluating boolean expressions in my head today :)

Reply:
Oh boy do I know that feeling, definitely been having that today too 😄